File: Demigods-Panel/src/main/scala/com/censoredsoftware/demigods/panel/DemigodsPanelPlugin.scala
package com.censoredsoftware.demigods.panel

import com.censoredsoftware.demigods.engine.mythos.{Mythos, MythosPlugin}
import com.censoredsoftware.censoredlib.trigger.Trigger
import org.bukkit.permissions.Permission
import org.bukkit.event.Listener
import com.censoredsoftware.demigods.engine.structure.Structure
import com.censoredsoftware.demigods.engine.deity.{Alliance, Deity}
import com.censoredsoftware.demigods.engine.item.DivineItem
import com.google.common.collect.{ImmutableSet, ImmutableCollection}
import org.bukkit.Bukkit
import org.bukkit.plugin.ServicePriority
import org.bukkit.inventory.ItemStack
import com.censoredsoftware.demigods.engine.item.DivineItem.Flag

class DemigodsPanelPlugin extends MythosPlugin
{
  override def onEnable(): Unit =
  {
    Bukkit.getServer.getServicesManager.register(classOf[Mythos], this, this, ServicePriority.Highest)
  }

  override def onDisable(): Unit =
  {}

  override def getTitle: java.lang.String = "Panel"

  override def getTagline: java.lang.String = "The Official Demigods Web Panel!"

  override def getAuthor: java.lang.String = "_Alex & HmmmQuestionMark"

  override def isPrimary: java.lang.Boolean = Boolean.box(x = false)

  override def allowSecondary(): java.lang.Boolean = Boolean.box(x = true)

  override def getIncompatible: Array[java.lang.String] = new Array[java.lang.String](0)

  override def useBaseGame(): java.lang.Boolean = Boolean.box(x = true)

  override def getDivineItems: ImmutableCollection[DivineItem] = ImmutableSet.of()

  override def getDivineItem(itemName: java.lang.String): DivineItem = Mythos.Util.getDivineItem(this, itemName)

  override def getDivineItem(itemStack: ItemStack): DivineItem = Mythos.Util.getDivineItem(this, itemStack)

  override def itemHasFlag(itemStack: ItemStack, flag: Flag): Boolean = Mythos.Util.itemHasFlag(this, itemStack, flag)

  override def getAlliances: ImmutableCollection[Alliance] = ImmutableSet.of()

  override def getAlliance(allianceName: java.lang.String): Alliance = Mythos.Util.getAlliance(this, allianceName)

  override def getDeities: ImmutableCollection[Deity] = ImmutableSet.of()

  override def getDeity(deityName: java.lang.String): Deity = Mythos.Util.getDeity(this, deityName)

  override def getStructures: ImmutableCollection[Structure] = ImmutableSet.of()

  override def getStructure(structureName: java.lang.String): Structure = Mythos.Util.getStructure(this, structureName)

  override def levelSeperateSkills(): java.lang.Boolean = Boolean.box(x = true)

  override def getListeners: ImmutableCollection[Listener] = ImmutableSet.of()

  override def getPermissions: ImmutableCollection[Permission] = ImmutableSet.of()

  override def getTriggers: ImmutableCollection[Trigger] = ImmutableSet.of()

  override def setSecondary(): Unit =
  {}

  override def lock(): Unit =
  {}
}

File: Demigods-Panel/Bukkit/src/main/java/com/censoredsoftware/demigods/panel/DemigodsPanelPlugin.java
package com.censoredsoftware.demigods.panel;

import com.censoredsoftware.censoredlib.helper.CommandManager;
import com.censoredsoftware.censoredlib.trigger.Trigger;
import com.censoredsoftware.demigods.engine.mythos.*;
import com.google.common.collect.ImmutableCollection;
import com.google.common.collect.ImmutableSet;
import org.bukkit.event.Listener;
import org.bukkit.inventory.ItemStack;
import org.bukkit.permissions.Permission;

public class DemigodsPanelPlugin extends MythosPlugin
{
	private static DemigodsPanelPlugin inst;

	public static DemigodsPanelPlugin inst()
	{
		return inst;
	}

	/**
	 * The Bukkit enable method.
	 */
	@Override
	public void onEnable()
	{
		inst = this;
		// TODO Connect to panel here.
	}

	/**
	 * The Bukkit disable method.
	 */
	@Override
	public void onDisable()
	{}

	@Override
	public String getTitle()
	{
		return "Web Panel";
	}

	@Override
	public String getTagline()
	{
		return "Official Panel for Demigods RPG.";
	}

	@Override
	public String getAuthor()
	{
		return "_Alex & HmmmQuestionMark";
	}

	@Override
	public Boolean isPrimary()
	{
		return false;
	}

	@Override
	public Boolean allowSecondary()
	{
		return true;
	}

	@Override
	public String[] getIncompatible()
	{
		return new String[] {};
	}

	@Override
	public Boolean useBaseGame()
	{
		return true;
	}

	@Override
	public ImmutableCollection<DivineItem> getDivineItems()
	{
		return ImmutableSet.of();
	}

	@Override
	public DivineItem getDivineItem(String itemName)
	{
		return Mythos.Util.getDivineItem(this, itemName);
	}

	@Override
	public DivineItem getDivineItem(final ItemStack itemStack)
	{
		return Mythos.Util.getDivineItem(this, itemStack);
	}

	@Override
	public boolean itemHasFlag(ItemStack itemStack, DivineItem.Flag flag)
	{
		return Mythos.Util.itemHasFlag(this, itemStack, flag);
	}

	@Override
	public ImmutableCollection<Alliance> getAlliances()
	{
		return ImmutableSet.of();
	}

	@Override
	public Alliance getAlliance(final String allianceName)
	{
		return Mythos.Util.getAlliance(this, allianceName);
	}

	@Override
	public ImmutableCollection<Deity> getDeities()
	{
		return ImmutableSet.of();
	}

	@Override
	public Deity getDeity(final String deityName)
	{
		return Mythos.Util.getDeity(this, deityName);
	}

	@Override
	public ImmutableCollection<Structure> getStructures()
	{
		return ImmutableSet.of();
	}

	@Override
	public Structure getStructure(final String structureName)
	{
		return Mythos.Util.getStructure(this, structureName);
	}

	public Boolean levelSeperateSkills()
	{
		return true;
	}

	public ImmutableCollection<Listener> getListeners()
	{
		return ImmutableSet.of();
	}

	public ImmutableCollection<Permission> getPermissions()
	{
		return ImmutableSet.of();
	}

	@Override
	public ImmutableCollection<CommandManager> getCommands()
	{
		return ImmutableSet.of();
	}

	public ImmutableCollection<Trigger> getTriggers()
	{
		return ImmutableSet.of();
	}

	@Override
	public void setSecondary()
	{}

	@Override
	public void lock()
	{}
}
